pep-0387.txt

```diff
@@ -112,6 +112,28 @@ Basic policy for backwards compatibility
   platforms).
 
 
+Soft Deprecation
+================
+
+A soft deprecation can be used when using an API must no longer be used
```

Comment:
- "must" implies commanding users that they are not allowed to write it in new code, even if they really want to
-  overall context is "no longer recommended, but perfectly fine to use in existing code as long as you don't expect new features"

IMHO this feels like a bit of a tonal oddity. Is this soft-deprecated functionality an actual concrete problem for new users who aren't interested in more features but just happen to really like this module anyway?

Maybe "should" is a better word to use here?

Reply:
For example optparse is mentioned down below, but not everyone considers that argparse is a proper replacement for optparse. Some people still like optparse, or just feel like they cannot use argparse without breaking their command-line parsing contract due to argparse's quirky design.

Reply:
So a Soft Deprecation is intended exclusively for cases where the code in question is objectively bad and anyone who uses it is assumed to be buggy then?

e.g. it cannot be used for getopt, because getopt is the correct (and only stdlib) way to do "parse arguments according to the high-level meta-conceptual semantics POSIX describes". Many people just prefer to use alternative argument resolution strategies when faced with ambiguity.

In that case, can it be clarified to point out that Soft Deprecation must not be used for code which only fell out of *popularity*?

e.g. don't mention:
> The API remains documented and tested, but will not be developed further (no enhancement).

because the lack of enhancements is non-topical. Instead mention:
> The API remains documented and tested, but has outstanding issues which will not be fixed.

Reply:
Bugs in soft deprecated APIs are fixing and will continue to be fixed.